src/domain/core/src/services/export_service.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::{Display, Formatter};
+use std::path::Path;
+
+use datafusion::dataframe::DataFrame;
+use internal_error::{BoxedError, InternalError};
+use thiserror::Error;
+
+use crate::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Service
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait ExportService: Send + Sync {
+    async fn export_to_fs(
+        &self,
+        df: DataFrame,
+        path: &Path,
```

Comment:
If we compare it to just extending `QueryService` with the `export_to_fs` method, is there any benefit in separate interface? It seems they travel together.

Reply:
I think logically it's a different things, `QueryService` is for preparing/reading data, and `ExportService` is for writing. When I started working on code, I was expecting something much more complex than just a few lines of code so created a separate service.
But with the current code state, having it separated or not is more matter of taste. I just left as it was.